### PR TITLE
Move some deps to devDependencies for TS RPC client

### PR DESCRIPTION
**Pull request by albrow**
Related to #428. We can take even more steps if needed; this is just the low-hanging fruit.

### CODE REVIEW

**Comment by albrow** [comment]
CI failed twice in a row with `error Incorrect integrity when fetching from the cache`. I'm looking into it...

**Comment by albrow** [comment]
Hard-coding the Yarn version in _.circleci/config.yml_ to `1.17` fixes the issue.

File: rpc/clients/typescript/package.json [hunk 1]
```diff
@@ -37,8 +30,6 @@
     },
     "homepage": "https://github.com/0xProject/0x-mesh/rpc/clients/typescript/README.md",
     "dependencies": {
-        "@0x/assert": "^2.1.2",
```
Comment (fabioberger) [inline rpc/clients/typescript/package.json]: The `@0x/assert` package is used in `ws_client.ts`, and is therefore not a `devDep`.

File: rpc/clients/typescript/package.json [hunk 2]
```diff
@@ -37,8 +30,6 @@
...
-        "@0x/dev-utils": "^2.2.6",
```
Comment (fabioberger) [inline rpc/clients/typescript/package.json]: `@0x/dev-utils` on the other hand is only used in tests and can therefore be moved. ✅

**Pull request merged**